### ISSUE TITLE
Crash on emulator 

### ISSUE THREAD

**Issue by trihung93** [jenzz/RxAppState#15]
Hello guys, 
First of all, Thank you so much for your library.
I got a crash when clicked button Pick Image again.

But I want to use the whole app, so I comment  getAppStateMonitor().start(); and  getAppStateMonitor().stop(); in MainActivity class . It work very well.
Tks.
```
05-12 21:49:40.447 3591-3591/com.jenzz.appstate.sample E/AndroidRuntime: FATAL EXCEPTION: main
                                                                         Process: com.jenzz.appstate.sample, PID: 3591
                                                                         java.lang.IllegalArgumentException: Receiver not registered: com.jenzz.appstate.internal.DefaultAppStateRecognizer$ScreenOffBroadcastReceiver@ebff3db
                                                                             at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1007)
                                                                             at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1330)
                                                                             at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:608)
                                                                             at com.jenzz.appstate.internal.DefaultAppStateRecognizer.stop(DefaultAppStateRecognizer.java:62)
                                                                             at com.jenzz.appstate.AppStateMonitor.stop(AppStateMonitor.java:54)
                                                                             at com.jenzz.appstate.sample.MainActivity$1.onClick(MainActivity.java:37)
                                                                             at android.view.View.performClick(View.java:5637)
                                                                             at android.view.View$PerformClick.run(View.java:22429)
                                                                             at android.os.Handler.handleCallback(Handler.java:751)
                                                                             at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                             at android.os.Looper.loop(Looper.java:154)
                                                                             at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```
```
package com.jenzz.appstate.sample;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import com.jenzz.appstate.AppStateMonitor;

import java.io.FileNotFoundException;
import java.io.InputStream;

import static android.content.Intent.ACTION_GET_CONTENT;

public class MainActivity extends AppCompatActivity {

  private static final int RESULT_CODE_PICK_IMAGE = 123;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    View pickImageButton = findViewById(R.id.pick_image);
    //noinspection ConstantConditions
    pickImageButton.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        getAppStateMonitor().stop();
        //pickImage();
        ActivityCompat.requestPermissions(MainActivity.this,
                new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},
                1);
      }
    });
  }

  @Override
  public void onRequestPermissionsResult(int requestCode,
                                         String permissions[], int[] grantResults) {
    switch (requestCode) {
      case 1: {

        // If request is cancelled, the result arrays are empty.
        if (grantResults.length > 0
                && grantResults[0] == PackageManager.PERMISSION_GRANTED) {

          // permission was granted, yay! Do the
          // contacts-related task you need to do.
        } else {

          // permission denied, boo! Disable the
          // functionality that depends on this permission.
          Toast.makeText(MainActivity.this, "Permission denied to read your External storage", Toast.LENGTH_SHORT).show();
        }
        return;
      }

      // other 'case' lines to check for other
      // permissions this app might request
    }
  }

  @Override
  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    super.onActivityResult(requestCode, resultCode, data);
    getAppStateMonitor().start();
    if (resultCode == RESULT_OK && requestCode == RESULT_CODE_PICK_IMAGE) {
      showImage(data.getData());
    }
  }

  private void pickImage() {
    Intent intent = new Intent(ACTION_GET_CONTENT);
    intent.setType("image/*");
    startActivityForResult(intent, RESULT_CODE_PICK_IMAGE);
  }

  private void showImage(Uri selectedImage) {
    try {
      InputStream imageStream = getContentResolver().openInputStream(selectedImage);
      Bitmap bitmap = BitmapFactory.decodeStream(imageStream);
      ImageView imageView = (ImageView) findViewById(R.id.image_result);
      //noinspection ConstantConditions
      imageView.setImageBitmap(bitmap);
    } catch (FileNotFoundException e) {
      // Ouch!
    }
  }

  private AppStateMonitor getAppStateMonitor() {return ((SampleApplication) getApplication()).getAppStateMonitor();}
}
```

**Comment by jenzz** [comment]
You seem to be calling `AppStateMonitor.stop()` before you have called `AppStateMonitor.start()`.
That results in a crash because stopping the monitor internally tries to unregister a `BroadcastReceiver` that hasn't been registered yet.

Make sure you are calling `AppStateMonitor.start()` either in `Application.onCreate()` or in `Activity.onCreate()`.

I might add some checks to ensure calling `stop()` multiple times does not result in a crash.

Please re-open if you still have problems.

**Comment by jenzz** [comment]
@chihung93 I have added a simple check to avoid crashing when calling `AppStateMonitor.stop()` multiple times.

It will be part of the next release.

**Comment by jenzz** [comment]
\- Released with [v3.0.1](https://github.com/jenzz/RxAppState/releases/tag/3.0.1) -

**Comment by chihung93** [comment]
@jenzz 👍  
Thank you so much
